### PR TITLE
SNOW-768195: [Client Side] Support GCS downscoped token for Snowpipe Streaming

### PR DESCRIPTION
**Pull request by sfc-gh-tzhang**

### CODE REVIEW

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm, Thanks! 
Please ensure test `testGCSDownscopedToken` is passing. It's failing currently.

File: src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestStage.java [hunk 1]
```diff
@@ -186,6 +185,7 @@ private void putRemote(String fullFilePath, byte[] data, int retryCount)
               .setStreamingIngestClientKey(this.clientPrefix)
               .setStreamingIngestClientName(this.clientName)
               .setProxyProperties(this.proxyProperties)
+              .setDestFileName(fullFilePath)
```
Comment (sfc-gh-japatel) [inline src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestStage.java:188]: Should this only be the file name instead of full path?

Comment (sfc-gh-tzhang) [reply src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestStage.java:188]: Not sure why in KC it's the file name, but I'm pretty sure we need to use fullFilePath here, and Spark Connector is using full path as well

**Comment by sfc-gh-tzhang** [comment]
> lgtm, Thanks! Please ensure test `testGCSDownscopedToken` is passing. It's failing currently.

Do you think we can give SNOWPIPE_SDK_GH_ACTION accountadmin role? How did you test KC?

Insufficient privileges to operate on user 'SNOWPIPE_SDK_GH_ACTION'

**Comment by sfc-gh-japatel** [comment]
> > lgtm, Thanks! Please ensure test `testGCSDownscopedToken` is passing. It's failing currently.
> 
> @sfc-gh-japatel Do you think we can give SNOWPIPE_SDK_GH_ACTION accountadmin role? How did you test KC?
> 
> Insufficient privileges to operate on user 'SNOWPIPE_SDK_GH_ACTION'

I ran the test with param enabled on the user and then re-ran it with param disabled. why is accountadmin needed.

**Comment by sfc-gh-tzhang** [comment]
> > > lgtm, Thanks! Please ensure test `testGCSDownscopedToken` is passing. It's failing currently.
> > 
> > 
> > @sfc-gh-japatel Do you think we can give SNOWPIPE_SDK_GH_ACTION accountadmin role? How did you test KC?
> > Insufficient privileges to operate on user 'SNOWPIPE_SDK_GH_ACTION'
> 
> I ran the test with param enabled on the user and then re-ran it with param disabled. why is accountadmin needed.?

How did you enable the parameter in test? Accountadmin is needed because I'm getting insufficient privilege

**Comment by sfc-gh-japatel** [comment]
Other way to do it is to set `alter account <accountname> set CHANGE_INTERNAL_PARAMS = true` from snowflake. this was I guess you can do it from your tests here. (making internal param external) I havent tried it though.

**Review by sfc-gh-rcheng — APPROVED** [review]

**Pull request merged**